examples/replset-docker/README.md

```diff
@@ -0,0 +1,81 @@
+# MongoDB Replication Example Using a Docker
+
+This [MongoDB replication](https://docs.mongodb.com/manual/replication/) example
+uses a [Docker engine](http://docker.com) to run replica set members. This platform
+do not provide any deployment automation and it is used mainly for developing and testing.
+
+## Getting Started
+
+You will need an Docker engine where you can run containers. If you want to avoid running all replset members on one host, you can also use [Docker Swarm](https://docs.docker.com/swarm/).
+
+## Example Working Scenarios
+
+This section describes how this example is designed to work.
+
+### Initial Deployment: 3-member Replica Set
+
+To create replica set with three members you can use this script:
+
+```bash
+function get_ip() {
+    local id="$1" ; shift
+    docker inspect --format='{{.NetworkSettings.IPAddress}}' ${id}
+}
+
+REPLICATION_ARGS="
+-e MONGODB_DATABASE=db
+-e MONGODB_USER=user
+-e MONGODB_PASSWORD=password
+-e MONGODB_ADMIN_PASSWORD=adminPassword
+-e MONGODB_REPLICA_NAME=rs0
+-e MONGODB_KEYFILE_VALUE=xxxxxxxxxxxx
+-e MONGODB_SMALLFILES=true"
+
+docker run -d --name replset0 ${BASE_ARGS} --hostname initialize -e MONGODB_REPLICA_MEMBERS="" $IMAGE_NAME run-mongod-replication
```

Comment:
> Does this mean that run-mongod-replication is supposed to inspect the hostname to decide whether to initialize the replica set?

Yes. Or it could be possible (maybe better?) to pass "initiate" parameter to `run-mongod-replication` script.

The main difference is that this container does not terminate after initialization.